### PR TITLE
hzn-818: Add factory classes for detectors 

### PR DESCRIPTION
**Pull request by cgorantla**
Add factory classes for detectors in order to support multiple instances being created 
* JIRA: http://issues.opennms.org/browse/hzn-818
* Bamboo: http://bamboo.internal.opennms.com:8085/browse/OPENNMS-ONMS784-3

### CODE REVIEW

File: opennms-provision/opennms-detector-simple/blueprint-detector.xml [hunk 1]
```diff
@@ -15,7 +15,8 @@
 	<service id="icmpDetector" ref="icmp"
```
Comment (j-white) [inline opennms-provision/opennms-detector-simple/blueprint-detector.xml]: Do we still need the detectors to be exposed if we have the factories?

File: opennms-provision/opennms-detector-simple/src/main/java/org/opennms/netmgt/provision/detector/factory/IcmpDetectorFactory.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+package org.opennms.netmgt.provision.detector.factory;
+
+import org.opennms.netmgt.provision.DetectorFactory;
+import org.opennms.netmgt.provision.detector.icmp.IcmpDetector;
+
+public class IcmpDetectorFactory implements DetectorFactory<IcmpDetector> {
+
+    @Override
+    public IcmpDetector createDetector() {
+        IcmpDetector detector = new IcmpDetector();
```
Comment (j-white) [inline opennms-provision/opennms-detector-simple/src/main/java/org/opennms/netmgt/provision/detector/factory/IcmpDetectorFactory.java]: We can get rid of these temporary variables and just `return new *;`

File: opennms-provision/opennms-detector-simple/blueprint-detector.xml (hunk 1 above)
Comment (cgorantla) [inline opennms-provision/opennms-detector-simple/blueprint-detector.xml]: yes, using them for getting the names in the Completer,  I can use the same factory there as well.

File: opennms-provision/opennms-detector-simple/blueprint-detector.xml [hunk 3]
```diff
@@ -43,7 +46,52 @@
...
+	</service>
+
+	<bean id="loopFactory"
+		class="org.opennms.netmgt.provision.detector.factory.LoopDetectorFactory">
+	</bean>
+	<service id="loopDetectorFactory" ref="loopFactory"
+		interface="org.opennms.netmgt.provision.DetectorFactory">
+		<service-properties>
+			<entry key="implementation"
+				value="org.opennms.netmgt.provision.detector.factory.LoopDetectorFactory" />
```
Comment (j-white) [inline opennms-provision/opennms-detector-simple/blueprint-detector.xml]: I would use the class name of the detector here instead.

**Comment by cgorantla** [comment]
review comments are taken care in https://github.com/OpenNMS/opennms/pull/852

**Pull request closed without merging**